Refresher/Patching/Patcher.cs

```diff
@@ -1,78 +1,164 @@
     {
-        foreach (string url in list)
+        BinaryReader reader = new(file);
+
+        //Get the length of the file (this operation can be costly, so lets cache it)
+        long fileLength = file.Length;
+
+        // Search for all instances of `http` in the binary
+        ReadOnlySpan<byte> http = "http"u8;
+
+        //Found positions of `http` in the binary
+        List<long> foundPositions = new();
+
+        //Get the length of the file (http.length also has a non-trivial cost, so we cache it)
+        long length = fileLength - http.Length;
+
+        //The buffer we read into
+        byte[] buf = new byte[http.Length];
+        long readPos = 0;
+        //While we are not at the end of the file
+        while (readPos < length)
         {
-            int offset = data.IndexOf(url, StringComparison.Ordinal);
-            if(offset < 1) continue;
+            //Set the position of the stream to the next one to check
+            reader.BaseStream.Position = readPos;
+
+            //Read into the buffer
+            int read = file.Read(buf);
+            //If we read less than the buffer size, we are at the end of the file
+            if (read < http.Length)
+                break;
 
-            return new PatchTargetInfo
+            bool equal = true;
+
+            //Check whether the original buffer is equal to the buffer we read
+            //NOTE: theres a slightly faster way of doing this with SIMD, but this is fine for now
+            //      see https://dev.to/antidisestablishmentarianism/c-simd-byte-array-compare-52p6
+            for (int i = 0; i < buf.Length; i++)
             {
-                Offset = offset,
-                Length = url.Length,
-            };
+                if (buf[i] == http[i])
+                    continue;
+
+                equal = false;
+                break;
+            }
+
+            //If they are equal, we found an instance of HTTP
+            if (equal)
+            {
+                //Mark the position of the found instance
+                foundPositions.Add(readPos);
+
+                //Skip the length of the buffer
+                readPos += buf.Length;
+            }
+            else
+            {
+                //Check the next position
+                readPos++;
+            }
         }
 
-        return null;
+        List<PatchTargetInfo> found = new();
+
+        bool tooLong = false;
+        foreach (long foundPosition in foundPositions)
+        {
+            int len = 0;
+
+            file.Position = foundPosition;
+
+            //Find the first null byte
+            while (reader.ReadByte() != 0)
+            {
+                len++;
+
+                if (len > 100)
+                {
+                    tooLong = true;
+                    break;
+                }
+            }
+
+            if (tooLong)
+            {
+                tooLong = false;
+                continue;
+            }
```

Comment:
Any particular reason for having `tooLong` outside the foreach like this? It's only set and read from within.

Reply:
This is something i do out of habit, as in a non-optimized setting, it means that a single variable is allocated on the stack once, rather than in every iteration, but *theoretically* the compiler would optimize them both to the `locals` section of the method in IL land anyway

Reply:
You are insane